File: monitor/src/main/java/com/kmdev/monitor/Application.java
package com.kmdev.monitor;

import io.vertx.reactivex.core.RxHelper;
import io.vertx.reactivex.core.Vertx;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

/**
 * Bootstrap application class.
 */
public final class Application {

    public static void main(String... args) {
        System.setProperty("vertx.logger-delegate-factory-class-name", "io.vertx.core.logging.SLF4JLogDelegateFactory");
        final Vertx vertx = Vertx.vertx();
        final Logger log = LoggerFactory.getLogger(Application.class.getName());

        Runtime.getRuntime().addShutdownHook(new Thread(() -> {
            log.info("Monitor module has been finished.");
            vertx.close();
        }));

        RxHelper.deployVerticle(vertx, new MonitorVerticle())
            .subscribe(val -> log.info("Monitor module has been started."), err -> System.exit(-1));
    }

}

File: monitor/src/main/kotlin/com/kmdev/monitor/Application.kt
package com.kmdev.monitor

import io.vertx.reactivex.core.RxHelper
import io.vertx.reactivex.core.Vertx
import org.slf4j.LoggerFactory

class Application

fun main(args: Array<String>) {

    System.setProperty("vertx.logger-delegate-factory-class-name", "io.vertx.core.logging.SLF4JLogDelegateFactory")

    val log = LoggerFactory.getLogger(Application::class.java.name)
    val vertx = Vertx.vertx()

    Runtime.getRuntime().addShutdownHook(Thread {
        log.info("Monitor module has been finished.")
        vertx.close()
    })

    RxHelper.deployVerticle(vertx, MonitorVerticle())
            .subscribe({ _ -> log.info("Monitor module has been started.") }, { _ -> System.exit(-1) })
}
